### PR TITLE
[MU4] Add files to create snap package of the latest stable version

### PR DESCRIPTION
**Pull request by pachulo**
I've been maintaining this snap for 2 years, back when I opened the first PR https://github.com/musescore/MuseScore/pull/3204

Back in the day @lasconic registered the `musescore` name on the store (https://snapcraft.io/musescore) and gave me rights to administer it. The snap has over 11k active installs right now.

I don't have time right now to maintain this anymore, so I thought that somebody from your organization could go on with this.

https://build.snapcraft.io/ can be used to create the snaps for the different archs and upload them to the store, or also https://launchpad.net/: the latest is what I was using, as I was only triggering manual builds when a new MuseScore version was available or a bundled package had a security update.

The latest version of the `snapcraft.yaml` has the following features:
- Doesn't have a pre-fixed version, will create a snap using the latest stable tag of this repo; that is, the newest tag that follows the pattern `v[0-9].[0-9].[0-9]$` (more info here https://forum.snapcraft.io/t/selective-checkout-check-out-the-tagged-release-revision-if-it-isnt-promoted-to-the-stable-channel/10617)
- Uses the `core18` base.
- Some interfaces need to be plugged manually: https://forum.snapcraft.io/t/snaping-musescore/704/9

And the following known issues:
- Musescore has to be compiled with `BUILD_WEBENGINE=off` because of this error https://musescore.org/en/comment/884393 that I couldn't work out.
- `Unable to load printer plugin "qtubuntu-print"` message appears when starting the application from the console.

Regarding everything else:
- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            # QT
+            - qtbase5-dev
+            - qttools5-dev
+            - qttools5-dev-tools
+            - qtwebengine5-dev
+            - qtscript5-dev
+            - libqt5xmlpatterns5-dev
+            - libqt5svg5-dev
+            - libqt5webkit5-dev
+            - qtquickcontrols2-5-dev
```
Comment (Jojo-Schmitz) [inline snap/snapcraft.yaml]: I believe "qml-module-qtquick-controls" is neede too, see e.g. https://musescore.org/en/node/301501#comment-980984

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            - libjack0
+            - libasyncns0
+            - libflac8
+            - qtwayland5
+            - libqt5help5
+            - libqt5sql5
+            # Palettes need these, see:
+            # https://forum.snapcraft.io/t/missing-dependency-for-musescore/14884
+            - qml-module-qtgraphicaleffects
+            - qml-module-qtquick-controls2
```
Comment (Jojo-Schmitz) [inline snap/snapcraft.yaml]: I believe "qml-module-qtquick-controls" is neede too, see e.g. https://musescore.org/en/node/301501#comment-980984

Comment (pachulo) [reply snap/snapcraft.yaml]: I've just added it! Thanks!

Comment (pachulo) [reply snap/snapcraft.yaml]: The version with this installed is on the beta/edge channel of the snap, but now this error appears when trying to load the `abc_import.qml` plugin: `QQmlComponent: Component is not ready`
Any ideas?

Comment (pachulo) [reply snap/snapcraft.yaml]: OK, the plugin loader showed this error: `Line 21: module "QtQuick.Dialogs" is not installed` when trying to execute the plugin, so at least package `qml-module-qtquick-dialogs` is also needed.

Comment (pachulo) [reply snap/snapcraft.yaml]: No I get these errors:
```
   line 47: module "Qt.labs.folderlistmodel" is not installed
   line 48: module "Qt.labs.settings" is not installed
   line 47: module "Qt.labs.folderlistmodel" is not installed
   line 48: module "Qt.labs.settings" is not installed
```
I guess we need more packages...

Comment (Jojo-Schmitz) [reply snap/snapcraft.yaml]: Those 2 are surely  needed indeed

Comment (pachulo) [reply snap/snapcraft.yaml]: OK, this is now working and on the snap stable channel. Thanks for the tips @Jojo-Schmitz !

File: snap/gui/musescore.desktop [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[Desktop Entry]
+Version=1.1
+Name=MuseScore
+GenericName=Music notation
+GenericName[de]=Notensatz
+GenericName[fr]=Notation musicale
+Comment=Create, play and print sheet music
+Comment[ru]=Визуальный редактор нотных партитур
+Comment[fr]=Gravure de partitions musicales
```
Comment (Marr11317) [inline snap/gui/musescore.desktop]: We would say "_En_gravure de partitions musicales"

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,192 @@
+name: musescore
```
Comment (Marr11317) [inline snap/snapcraft.yaml]: I am not sure anymore what is the rule for uppercasing

File: snap/gui/musescore.desktop [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[Desktop Entry]
```
Comment (shoogle) [inline snap/gui/musescore.desktop:1]: Just a quick note for anybody who might decide to pick this up. This file is basically the same as [build/Linux+BSD/mscore.desktop.in](https://github.com/musescore/MuseScore/blob/102fe1c4876ac0e410b3d61883784536bd780efa/build/Linux%2BBSD/mscore.desktop.in) with just a few slight (but important) differences. The two files should probably be combined into a single file, with those differences inserted at build time by CMake.

File: snap/snapcraft.yaml (hunk 4 above)
Comment (Jojo-Schmitz) [reply snap/snapcraft.yaml:1]: Still "MuseScore"

File: snap/gui/musescore.desktop (hunk 3 above)
Comment (Jojo-Schmitz) [reply snap/gui/musescore.desktop:9]: Same 'mistake' as in build/Linux+BSD/mscore.desktop.in

File: snap/gui/musescore.desktop [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+Comment=Create, play and print sheet music
+Comment[ru]=Визуальный редактор нотных партитур
+Comment[fr]=Gravure de partitions musicales
+Exec=musescore.mscore %F
+Icon=${SNAP}/usr/share/icons/hicolor/scalable/apps/mscore.svg
+StartupNotify=true
+Terminal=false
+Type=Application
+Categories=Qt;Audio;Sequencer;Midi;AudioVideoEditing;Music;AudioVideo;
+MimeType=application/x-musescore;application/x-musescore+xml;application/vnd.recordare.musicxml;application/vnd.recordare.musicxml+xml;
```
Comment (Jojo-Schmitz) [inline snap/gui/musescore.desktop:16]: this here is much shorter than that from build/Linux+BSD/mscore.desktop.in, why?

**Comment by Jojo-Schmitz** [comment]
(But I've included an updated version of it in PR #9000, so you could provide a snap package of a ' 3.7.0' version)

**Pull request closed without merging**